src/parser/jobs/drg/modules/RotationWatchdog.js

```diff
@@ -0,0 +1,176 @@
+import {Trans, Plural, i18nMark} from '@lingui/react'
+import React from 'react'
+import {Accordion, Message} from 'semantic-ui-react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import Rotation from 'components/ui/Rotation'
+import ACTIONS, {getAction} from 'data/ACTIONS'
+import Module from 'parser/core/Module'
+
+const VALID_ROTATIONS = [
+	[ // The normal rotation: HT -> DoT -> FT
+		ACTIONS.HEAVY_THRUST.id,
+		ACTIONS.IMPULSE_DRIVE.id,
+		ACTIONS.DISEMBOWEL.id,
+		ACTIONS.CHAOS_THRUST.id,
+		ACTIONS.WHEELING_THRUST.id,
+		ACTIONS.FANG_AND_CLAW.id,
+		ACTIONS.TRUE_THRUST.id,
+		ACTIONS.VORPAL_THRUST.id,
+		ACTIONS.FULL_THRUST.id,
+		ACTIONS.FANG_AND_CLAW.id,
+		ACTIONS.WHEELING_THRUST.id,
+	],
+	[ // The inverse rotation: HT -> FT -> DoT
+		ACTIONS.HEAVY_THRUST.id,
+		ACTIONS.TRUE_THRUST.id,
+		ACTIONS.VORPAL_THRUST.id,
+		ACTIONS.FULL_THRUST.id,
+		ACTIONS.FANG_AND_CLAW.id,
+		ACTIONS.WHEELING_THRUST.id,
+		ACTIONS.IMPULSE_DRIVE.id,
+		ACTIONS.DISEMBOWEL.id,
+		ACTIONS.CHAOS_THRUST.id,
+		ACTIONS.WHEELING_THRUST.id,
+		ACTIONS.FANG_AND_CLAW.id,
+	],
+	[ // The two-target rotation: HT -> DoT -> more DoT (okay stop DoT)
+		ACTIONS.HEAVY_THRUST.id,
+		ACTIONS.IMPULSE_DRIVE.id,
+		ACTIONS.DISEMBOWEL.id,
+		ACTIONS.CHAOS_THRUST.id,
+		ACTIONS.WHEELING_THRUST.id,
+		ACTIONS.FANG_AND_CLAW.id,
+		ACTIONS.IMPULSE_DRIVE.id,
+		ACTIONS.DISEMBOWEL.id,
+		ACTIONS.CHAOS_THRUST.id,
+		ACTIONS.WHEELING_THRUST.id,
+		ACTIONS.FANG_AND_CLAW.id,
+	],
+]
+
+const AOE_ACTIONS = [
+	ACTIONS.DOOM_SPIKE.id,
+	ACTIONS.SONIC_THRUST.id,
+]
+
+const ROTATION_GCD_COUNT = 11
+const GCD_LENGTH = 2500
```

Comment:
I mean I know it's not hyper accurate, but is it worth using `gcd.getEstimate`?

Reply:
I considered it and I can make the change if you feel strongly, but I figured a generic 2.5s was fine here since it's just a "did we enter downtime" buffer.

Reply:
Can resolve if it becomes troublesome down the track.